### PR TITLE
Image caching

### PR DESCRIPTION
**Pull request by alxhoff**
@PhilippvK  I have moved the renderer out of the binding and back into init, it is no longer recreated but the loaded images in main seems to work. Give it a test and see if we run into our original errors before you did the context fixes. If we don't have errors then this code should just need some basic polishing and documenting.

### CODE REVIEW

**Comment by alxhoff** [comment]
I am not getting the context errors as far as I can see from my ~2 minutes of random testing

**Comment by PhilippvK** [comment]
That was fast! 

One quick note: The prefix of the DrawScaledImage funktion should be `x` instead of `v` no as you do not return void anymore.

**Comment by alxhoff** [comment]
> That was fast!
> 
> One quick note: The prefix of the DrawScaledImage funktion should be `x` instead of `v` no as you do not return void anymore.

yep I saw this and will overhaul them all at the end

**Comment by PhilippvK** [comment]
I tried out the branch and benchmarked it (in the emulator).

Drawing performance using jpg just got a little bit better but especially BMP files are way better now!

To test the drawing context related stuff I have to kickoff my docker toolchain again which was broken the last time i used it... (Sorry if I need some time)

**Comment by PhilippvK** [comment]
Hmmm, like already expected, it does in fact break my docker toolchain which relies heavily on X11 Forwarding. All I can see is a white screen as the renderer is associated with the wrong thread. If would add drawing statements in the initDrawing function they would appear..

**Comment by alxhoff** [comment]
looks like i'll have to do some sort of sorcery with loaded images and reloading them when the GL context is rebound zzzzz. I'll look into it tomorrow

**Comment by PhilippvK** [comment]
I could live without the code working in these special cases as I do not use the toolchain anymore. But in the end it would be the the best idea to to every 100% thread save.

File: lib/Gfx/TUM_Draw.c [hunk 1]
```diff
@@ -808,6 +813,30 @@ int vInitDrawing(char *path) // Should be called from the Thread running main()
 
 	atexit(SDL_Quit);
 
+
+    //TEST
+    FILE *my_img = fopen("../resources/freertos.jpg", "r");
```
Comment (alxhoff) [inline lib/Gfx/TUM_Draw.c]: @PhilippvK  I think this sort of process will be the best caching I can do. I will just need to rerun `SDL_CreateTextureFromSurface` for each image every time the context is rebound.

**Comment by alxhoff** [comment]
@PhilippvK this should do it. I will just add in warning for using non-loaded and scaled image calls but hopefully this is stable. Let me know if it's stable then I will rebase into backend and main commits and merge

**Comment by alxhoff** [comment]
@PhilippvK  in the main I freed the first image then tried to draw it, this spams the terminal with an error message. it's by design and I will simplify the demo code in the main before merging.

**Comment by PhilippvK** [comment]
I am very impressed. Improvements are definitely recognizable (f.e. when Drawing the Logo >10 times).

In my VM I sometimes experience crashed when the Emulator is launching. If have the feeling that that it gets worse the more graphics are initialized in the `main()` but I would ignore it for now. Happened for the 4th time in 2 week now.

**Review by PhilippvK — APPROVED** [review]
works fine for me!

**Comment by PhilippvK** [comment]
great that you have figured it out to implement this without breaking my docker toolchain :D

**Comment by alxhoff** [comment]
Awesome. I also refactored a lot of the code and made a couple of quick videos about it.

**Pull request merged**